### ISSUE TITLE
Ability to rename additionalProperties keys

### ISSUE THREAD

**Issue by ctiaffay-conserto** [swagger-api/swagger-ui#5713]
### Content & configuration

Swagger/OpenAPI definition:
 ```js
{
    type: 'object',
    additionalProperties: {
        $ref: '#/components/schemas/item',
    },
}
 ```

 Rendered schema in swagger UI :
 ```js
 {
    "additionalProp1": {},
    "additionalProp2": {},
    "additionalProp2": {}
}
 ```

### Problem :
Here, we're totally losing the context of the objects keys. And users of my api don't know at all what is "additionalProp1" means.
In my case, each objects keys are email address and this important information is totally lost.

### Possible solution :
If we could rename the automatic key prefix, it would be great ! :

 Rendered schema in swagger UI
 ```js
 {
    "dynamicEmailAddress1": {},
    "dynamicEmailAddress2": {},
    "dynamicEmailAddress3": {}
}
 ```

But I don't know if it's doable to insert a new property somewhere (like `additionalPropertiesPrefix: 'dynamicEmailAddress'` in the open api schema, so, sorry in advance if it's not the case.

Thanks

**Comment by webron** [comment]
Currently, there's no way to control it, but why not just provide your own example that would be more meaningful?

**Comment by ctiaffay-conserto** [comment]
Yeah seems like the only solution :( But it would mean to hardcode the value structure instead of letting it be auto generated from schema, and I think it is worse.
Thanks

**Comment by tmoreno** [comment]
Hi!

I have the same issue, swagger-ui generates examples in this way

`{
  "additionalProp1": {
    "code": 11,
    "text": "message text sample"
  },
  "additionalProp2": {
    "code": 11,
    "text": "message text sample"
  },
  "additionalProp3": {
    "code": 11,
    "text": "message text sample"
  }
}
`

And I would like to specify a value for the key:

`
{
  "randomUUID1": {
    "code": 11,
    "text": "message text sample"
  }
}
`

Thanks!

**Comment by ddweber** [comment]
Any updates on this topic?

**Comment by Pryftan** [comment]
Would love to see some improvements here. Auto-generated examples from schema are great for everything but dictionaries, where this issue forces users to create entirely custom examples since the automatically generated additionalProp#s are not at all descriptive.

**Comment by twankamp** [comment]
I ran into this issue too and created #9739 to resolve this using the standard propertyNames from json schema 2020-12.

Now if you want exactly 2 examples with names dynamicEmailAddress1 and dynamicEmailAddress2:
```js
{
   "dynamicEmailAddress1": {},
   "dynamicEmailAddress2": {}
}
```
You would write:
```yml
{
   propertyNames:
      examples:
      - 'dynamicEmailAddress1'
      - 'dynamicEmailAddress2'
   additionalProperties:
      $ref: "#/components/schemas/Object"
}
```
The referred object itself could be a full-fledged schema for which the samples will be generated as normal. So the examples are just for the property names.

Alternatively, if the property names should adhere to email format you can also:
```yml
{
   propertyNames:
      format: email
   additionalProperties:
      $ref: "#/components/schemas/Object"
}
```
To get:
```js
{
   "user1@example.com": {},
   "user2@example.com": {},
   "user3@example.com": {}
}
```

Hope it gets accepted and merged.
